src/components/OSCALControlProse.js

```diff
@@ -267,22 +268,45 @@ function getParameterValueSegment(
     parameterId
   );
 
+  if (parameterValue) {
+    if (!constraintsDisplay.length) {
+      return (
+        <ParamValue component="span" key={`param-value-key-${key}`}>
+          {parameterValue}
+        </ParamValue>
+      );
+    }
+    return (
+      <SegmentTooltipWrapper
+        constraintsDisplay={constraintsDisplay}
+        key={`segment-wrapper-key-${key}`}
+      >
+        <ParamValue component="span" key={`param-value-key-${key}`}>
+          {parameterValue}
+        </ParamValue>
+      </SegmentTooltipWrapper>
+    );
+  }
+  const parameterLabelText = getParameterLabelText(
+    parameters,
+    parameterId,
+    modificationSetParameters
+  );
   if (!constraintsDisplay.length) {
     return (
-      <ParamValue component="span" key={`param-value-key-${key}`}>
-        {parameterValue}
-      </ParamValue>
+      <ParamLabel component="span" key={`param-label-key-${key}`}>
+        {parameterLabelText}
+      </ParamLabel>
     );
   }
-
   return (
     <SegmentTooltipWrapper
-      constraintsDisplay={constraintsDisplay}
+      constraintDisplay={constraintsDisplay}
       key={`segment-wrapper-key-${key}`}
     >
-      <ParamValue component="span" key={`param-value-key-${key}`}>
-        {parameterValue}
-      </ParamValue>
+      <ParamLabel component="span" key={`param-label-key-${key}`}>
```

Comment:
These `ParamLabels` seem very similar. Could they be extracted to a function?

Something like
```
const paramLabel = (text, key) => { return <ParamLabel> }
```

Same thing for the `ParamValue`, its very little duplication but it would not take much to fix.

Reply:
That makes sense. I'll try doing that

Reply:
@tuckerzp My most recent commit has addressed this, let me know if it is what you were thinking!

Reply:
That is what I meant! Looking at the file as a whole though, we could make it even more general to the file itself. For example, all `<ParamLabel>` tags are using a `parameterLabelText` & a `key`. Would it be much more effort to generalize both the functions you created to reduce duplication in the entire file? You may want to rename the function so it is not almost exactly similar to the type declaration at the top of the file though (`paramLabel` vs `ParamLabel`) .

Reply:
I can do that!